### PR TITLE
Add promotion results, received via REST API, to deployment

### PR DESCRIPTION
**Pull request by maheshkelkar**
The issue 19 implemented the REST APIs and this code processes
the promotion status updates

References #82.

### CODE REVIEW

File: src/main/groovy/deploydb/resources/DeploymentResource.groovy [hunk 1]
```diff
@@ -26,9 +32,35 @@ import javax.ws.rs.core.Response
...
+
+        /**
+         *  Valid State transitions for deployment are:
+         *  NOT_STARTED -> STARTED
+         *  NOT_STARTED -> COMPLETED
+         *  NOT_STARTED -> FAILED
+         *  STARTED -> COMPLETED
+         *  STARTED -> FAILED
+         */
+        deploymentStatusTransitionPairs = new HashSet<>()
```
Comment (rtyler) [inline src/main/groovy/deploydb/resources/DeploymentResource.groovy]: The resource really does seem like the wrong place to defines these. Why wouldn't this be related to the Deployment model itself?

File: src/main/groovy/deploydb/resources/DeploymentResource.groovy [hunk 2]
```diff
@@ -26,9 +32,35 @@ import javax.ws.rs.core.Response
...
+        deploymentStatusTransitionPairs.add(Pair.of(Status.NOT_STARTED, Status.FAILED))
+        deploymentStatusTransitionPairs.add(Pair.of(Status.STARTED, Status.COMPLETED))
+        deploymentStatusTransitionPairs.add(Pair.of(Status.STARTED, Status.FAILED))
+
+        /**
+         *  Valid State transitions for PromotionResult are:
+         *  STARTED -> SUCCESS
+         *  STARTED -> FAILED
+         */
+        promotionResultStatusTransitionPairs = new HashSet<>()
```
Comment (rtyler) [inline src/main/groovy/deploydb/resources/DeploymentResource.groovy]: This mapping should also be in the PromotionResult model itself

**Comment by maheshkelkar** [comment]
@rtyler I have incorporated your comments. Do you want me to go ahead and merge this?

File: src/main/groovy/deploydb/resources/DeploymentResource.groovy [hunk 3]
```diff
@@ -102,12 +108,51 @@ public class DeploymentResource {
         }
 
         /**
-         *  FIXME: valid deploymentUpdateMapper.status are (Started, Completed, Failed)
+         *  Check for valid status transitions. Throw exception if not found
+         */
+        if (!Deployment.class.deploymentStatusTransitionPairs.contains(
```
Comment (rtyler) [inline src/main/groovy/deploydb/resources/DeploymentResource.groovy]: I do not believe you need to refer to `Deployment.class`, `Deployment` should be sufficient

**Pull request merged**